### ISSUE TITLE
[a11y]: List Page_Screen reader announces the entire table content after pagination

### ISSUE THREAD

**Issue by anamhussain94** [carbon-design-system/carbon#21054]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.96.0

### Automated testing tool and ruleset

WCAG Mapping: 4.1.2

### Assistive technology

MacOS voiceover

### Description

When a user activates the next or previous button in pagination, the table updates with new content. However, screen readers automatically read out the entire updated table, makes it harder for screen reader users to stay oriented.

Avoid announcing the entire table. Instead, use a brief message like “Items 21–30 of 88” to confirm the update, and ensure it is announced by screen readers.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-pagination--overview

### Steps to reproduce

When a user activates the next or previous button in pagination, the table updates with new content. However, screen readers automatically read out the entire updated table, makes it harder for screen reader users to stay oriented.

Avoid announcing the entire table. Instead, use a brief message like “Items 21–30 of 88” to confirm the update, and ensure it is announced by screen readers.

**Steps to reproduce**
1. Navigate to carbon/design [Pagination](https://react.carbondesignsystem.com/?path=/docs/components-pagination--overview&addons=1&panelRight=0&selectedKind=Accordion&selectedStory=Default&stories=1)
2. Use voiceover or JAWS and click on the pagination prev and next button. The screen reader should announce no. of items.

<img width="813" height="184" alt="Image" src="https://github.com/user-attachments/assets/8c5151fe-c956-47d4-9581-0de1d29f8b00" />

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
I discussed this issue with the accessibility team, and indeed the best approach would be for the screen reader to announce a message such as “Items 1–10 of 103 items” to confirm the update when moving to next or previous.

<img width="804" height="143" alt="Image" src="https://github.com/user-attachments/assets/44dc6aa8-fc36-4c55-9021-08e8d9a78d9c" />
